vars/runBenchmarkTestScript.groovy

```diff
@@ -0,0 +1,100 @@
+ * @param args.bundleManifest <required> - OpenSearch bundle manifest url.
+ * @param args.insecure <required> - Force the security of the cluster to be disabled, default is false.
+ * @param args.workload <required> - Name of the workload that OpenSearch Benchmark should run, default is nyc_taxis.
+ * @param args.singleNode <required> - Create single node OS cluster, default is true.
+ * @param args.minDistribution <required> - Use min distribution of OpenSearch for cluster, default is false.
+ * @param args.useFiftyPercentHeap <required> - Use 50 percent of physical memory as heap, default is false.
+ * @param args.captureNodeStat <required> - Make opensearch-benchmark to capture node stats during run, default is false
+ * @param args.suffix <optional> - Suffix to be added to stack name for benchmark test.
+ * @param args.managerCount <optional> - Number of manager nodes in multi-node cluster, default is 3.
+ * @param args.dataCount <optional> - Number of data nodes in multi-node cluster, default is 2.
+ * @param args.clientCount <optional> - Number of client nodes in multi-node cluster, default is 0.
+ * @param args.ingestCount <optional> - Number of ingest nodes in multi-node cluster, default is 0.
+ * @param args.mlCount <optional> - Number of ml nodes in multi-node cluster, default is 0.
+ * @param args.workloadParams <optional> - Additional parameters for benchmark workload type.
+ * @param args.dataStorage <optional> - Data node ebs storage size, default is 100G.
+ * @param args.mlStorage <optional> - Ml node ebs storage size, default is 100G.
+ * @param args.jvmSysProps <optional> - Custom JVM properties to be set for OS cluster.
+ * @param args.userTag <optional> - Additional metadata tags to be added to benchmark run metrics.
+ */
+void call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.bundleManifest))
+
+    config_name = isNullOrEmpty(args.config) ? "config.yml" : args.config
+    benchmark_config = "benchmark.ini"
+    withCredentials([string(credentialsId: 'jenkins-aws-account-public', variable: 'AWS_ACCOUNT_PUBLIC'),
+                     string(credentialsId: 'jenkins-artifact-bucket-name', variable: 'ARTIFACT_BUCKET_NAME')]) {
+        withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
+            s3Download(file: "config.yml", bucket: "${ARTIFACT_BUCKET_NAME}", path: "${BENCHMARK_TEST_CONFIG_LOCATION}/${config_name}", force: true)
+            s3Download(file: "benchmark.ini", bucket: "${ARTIFACT_BUCKET_NAME}", path: "${BENCHMARK_TEST_CONFIG_LOCATION}/${benchmark_config}", force: true)
+
+            /*Added sleep to let the file get downloaded first before write happens. Without the sleep the write is
+            happening in parallel to download resulting in file not found error
+             */
+            sleep(5)
```

Comment:
did you check if there is a parameter to wait until this completes? Maybe on jenkins side or `s3Download` steps?